elastic/kibana
# [7.5] [Discover] fix histogram min interval

## Summary

This commit fix the missing bars on the histogram of Discover.
Discover is passing a minInterval parameter in milliseconds that is not
the the minimum interval in milliseconds between the buckets. Calendar intervals
are not fixed and this value change depending on time range and the time zone.
To avoid patching the elastic-chart and upgrading the dependency we applied a temporary patch
to the histogram code. For the feature version this patch will be removed
and a correct minInterval computation will be done directly in elastic-charts.

fix #52152

**to test:**
add some data that spans multiple years (possibly leap years, last one was 2016)
```
node scripts/makelogs.js -c 1m -d 2200/1
// or import the esArchived dataset
node scripts/es_archiver.js --es-url elastic:changeme@localhost:9275 --kibana-url http://elastic:changeme@localhost:5675 load long_window_logstash
```
create the index pattern for that index: `logstash-*` or `long_window_logstash*`
check the behaviour on histogram when the timerange spans several years (including leap years) with different interval configuration. The issue is visible when:
- visualizing a timerange with monthly intervals that contains data in the February bucket
**timezone**: `all`
**time range**: `Nov 1, 2017 @ 01:00:00.000 - Mar 21, 2018 @ 02:00:00.000`
**intervals**: `Monthly`

- visualizing a timerange with a DST time change within the range. The DST change needs to be a positive change like when the time is moved forward (= that day last less than 24 hours) like in the following situation:
**timezone**: `Europe/Berlin` (doesn't happen with UTC timezone)
**time range**: `Mar 1, 2018 @ 00:00:00.000 - May 1, 2018 @ 00:00:00.000` 
**intervals**: `all except Monthly and Yearly`

- visualizing a timerange with a DST time change with `auto` interval on a fixed/multi days interval:
**timezone**: `Europe/Berlin`
**time range**: `Last 15 years - now`
**intervals**: every interval smaller than month (be sure that the scaled interval is scaled to 30days)

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.


### For maintainers
